app/src/androidTest/java/com/github/sdp_begreen/begreen/firebase/FirebaseDBTest.kt

```diff
@@ -253,4 +253,192 @@ class FirebaseDBTest {
         }
     }
 
+    @Test
+    fun followThrowsIllegalArgumentExceptionWhenGivenBlankUserIdAsFirstArg() {
+        assertThrows(IllegalArgumentException::class.java) {
+            runBlocking {
+                FirebaseDB.follow("", "Non existent ID")
+            }
+        }
+    }
+
+    @Test
+    fun followDoesNothingWhenGivenNonExistentUserId() {
+        runBlocking {
+            FirebaseDB.follow("Non existent ID", "Non existent ID")
+        }
+    }
+
+    @Test
+    fun followCorrectlyAddsFollowerIdToFollowedUserAndFollowedIdToFollowerUser() {
+        val user1 = User("user1 id", 5)
+        val user2 = User("user2 id", 3)
+
+        // Add both users to the DB
+        runBlocking {
+            FirebaseDB.addUser(user1, user1.id)
+            FirebaseDB.addUser(user2, user2.id)
+        }
+
+        // Check that both users have no followers and following users
+        assertThat(user1.followers, nullValue())
+        assertThat(user1.following, nullValue())
+        assertThat(user2.followers, nullValue())
+        assertThat(user1.following, nullValue())
```

Comment:
I guess it is a typo: shouldn't be "assertThat(user2.following, nullValue())" here?

Reply:
You're totally right, well found!